python/lsst/ip/isr/crosstalk.py

```diff
@@ -0,0 +1,205 @@
+        Median background level.
+    """
+    mask = mi.getMask()
+    stats = lsst.afw.math.StatisticsControl()
+    stats.setAndMask(mask.getPlaneBitMask(badPixels))
+    return lsst.afw.math.makeStatistics(mi, lsst.afw.math.MEDIAN, stats).getValue()
+
+
+def subtractCrosstalk(exposure, coeffs, badPixels=["BAD"], minPixelToMask=45000, crosstalkStr="CROSSTALK"):
+    """Subtract the intra-CCD crosstalk from an exposure
+
+    We set the mask plane indicated by ``crosstalkStr`` in a target amplifier
+    for pixels in a source amplifier that exceed `minPixelToMask`. Note that
+    the correction is applied to all pixels in the amplifier, but only those
+    that have a substantial crosstalk are masked with ``crosstalkStr``.
+
+    The uncorrected image is used as a template for correction. This is good
+    enough if the crosstalk is small (e.g., coefficients < ~ 1e-3), but if it's
+    larger you may want to iterate.
+
+    Parameters
+    ----------
+    exposure : `lsst.afw.image.Exposure`
+        Exposure for which to subtract crosstalk.
+    coeffs : `numpy.ndarray`
+        Matrix of crosstalk coefficients. ``coeffs[i][j]`` is the fraction
+        of the j-th amp present on the i-th amp.
+    badPixels : `list` of `str`
+        Mask planes to ignore.
+    minPixelToMask : `float`
+        Minimum pixel value in source amplifier for which to set
+        ``crosstalkStr`` mask plane in target amplifier.
+    crosstalkStr : `str`
+        Mask plane name for pixels greatly modified by crosstalk.
+    """
+    mi = exposure.getMaskedImage()
+    mask = mi.getMask()
+    bg = calculateBackground(mi, badPixels)
+
+    ccd = exposure.getDetector()
+    numAmps = len(ccd)
+    assert coeffs.shape == (numAmps, numAmps)
+
+    # Set the crosstalkStr bit for the bright pixels (those which will have significant crosstalk correction)
+    crosstalkPlane = mask.addMaskPlane(crosstalkStr)
+    lsst.afw.display.getDisplay().setMaskPlaneColor(crosstalkStr, lsst.afw.display.MAGENTA)
+    footprints = lsst.afw.detection.FootprintSet(mi, lsst.afw.detection.Threshold(minPixelToMask))
+    footprints.setMask(mask, crosstalkStr)
+    crosstalk = mask.getPlaneBitMask(crosstalkStr)
+
+    subtrahend = mi.Factory(mi.getBBox())
+    subtrahend.set((0, 0, 0))
+    for ii, iAmp in enumerate(ccd):
+        iImage = subtrahend.Factory(subtrahend, iAmp.getBBox())
+        for jj, jAmp in enumerate(ccd):
+            if ii == jj:
+                assert coeffs[ii, jj] == 0.0
+            if coeffs[ii, jj] == 0.0:
+                continue
+
+            jImage = extractAmp(mi, jAmp, iAmp.getReadoutCorner())
+            jImage.getMask().getArray()[:] &= crosstalk  # Remove all other masks
+            jImage -= bg
+
+            iImage.scaledPlus(coeffs[ii, jj], jImage)
+
+    # Set crosstalkStr bit only for those pixels that have been significantly modified
+    # (not necessarily those that are bright originally).
+    mask.clearMaskPlane(crosstalkPlane)
```

Comment:
I'm missing something here.  Doesn't this clear the mask plane?  Where do we set the crosstalk bit?